### DISCUSSION TITLE
Android Client tight coupling with AppCompatActivity.

### DISCUSSION THREAD

**Discussion by being-eyram** [speechly/speechly#204] [Q&A]
My experience using **Speechly Client** for **Android** has been great so far.
The ease of setting up within few minutes is unmatched in my experience with SDKs so far. 
However, there are certain design decisions that I don't understand.
For example, instantiating a **Client** requires an **Activity**, which in my case makes dependency injection difficult. 
In my application, I'd like to use the **Client** in a ViewModel and this is not possible since the ViewModel is not supposed to know about the Activity/ Fragment. 
My question is, is this design decision by choice or as  a result of constraints within the Android Library?

**Comment by langma** [comment]
Hi!

Thanks for the good feedback :)

You have a good point that the Activity based instantiation is not that convenient for your use case. Would you have an example of a ViewModel based application that I could look into? We might achieve this with some small changes to the Client implementation, but I'd like to understand the technical implications thoroughly first.

**Reply by being-eyram** [reply]
Unfortunately, I don't have any examples that I can share with you. 
But I'll keep looking though.
